core/src/main/java/org/opensearch/sql/expression/function/FunctionDSL.java

```diff
@@ -137,6 +137,52 @@ public String toString() {
     };
   }
 
+  /**
+   * Implementation of a function that takes two argument, returns a value, and
+   * requires FunctionProperties to complete.
+   *
+   * @param function {@link ExprValue} based binary function.
+   * @param returnType return type.
+   * @param args1Type first argument type.
+   * @param args2Type second argument type.
+   * @return Binary Function Implementation.
+   */
+  public static SerializableFunction<FunctionName, Pair<FunctionSignature, FunctionBuilder>>
+      implWithProperties(
```

Comment:
You can simplify two-args-function `impl` wrapper to use this one now. See (copy) it from https://github.com/Bit-Quill/opensearch-project-sql/blob/integ-datetime-calculations/core/src/main/java/org/opensearch/sql/expression/function/FunctionDSL.java

Reply:
Fixed in 2517b31f4ed5688dcaa7baa78aa4a025a71a28a4